### PR TITLE
Add typhoon_h480_livox vehicle + livox lidar sim capability

### PR DESCRIPTION
**Pull request by ahmed-elsaharti**
**Description**
This PR:
Closes: https://github.com/Airtonomy/flight-gazebo-sim/issues/20
Adds the ability to include any of the [livox lidar simulations](https://github.com/Livox-SDK/livox_laser_simulation) with our gazebo sim.

This could be tested by selecting the typhoon_h480_livox as a vehicle while launching a gazebo PX4 sim (ie: `make px4_sitl gazebo_typhoon_h480_livox` while using https://github.com/Airtonomy/flight-px4-autopilot/tree/ae/typhoon-livox )

**For Developers**
Make sure that you have done all of the steps necessary to merge a PR:
- [x] My code follows the coding style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated and created tests accordingly.
- [ ] My PR is linked to an issue.
- [ ] The PR fully addresses the issue, or follow up issues are created.

**For Reviewers**
- [ ] Are the tests that need to be updated changed?
- [ ] Does the code conform to the standards of the project?
- [ ] Is the design consistent with the architecture of the project?

**Additional context**
Add any other context or screenshots about the pull request here.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 # Set c++11 or higher
 include(EnableC++XX)
 # Set c11
-set(CMAKE_C_STANDARD 11)
```
Comment (stevenmorphet-at) [inline CMakeLists.txt:28]: How that sneak in?

Comment (stevenmorphet-at) [reply CMakeLists.txt:28]: We should probably check other projects for C++11 -v- 17

Comment (ahmed-elsaharti) [reply CMakeLists.txt:28]: The livox plugin requires c++17 to compile. Took a look at our implementation and found that c++17 doesn't break anything so it was safe to switch over

Comment (ahmed-elsaharti) [reply CMakeLists.txt:28]: To be fair i didn't test c++11 with livox, I decided to trust their Cmakelists

Comment (stevenmorphet-at) [reply CMakeLists.txt:28]: That's fine, I thought we we doing everything using the 17 standard?! The 11 standard is a little long in the tooth

Comment (ahmed-elsaharti) [reply CMakeLists.txt:28]: Agreed, gazebo and PX4 are expected to have lurking c++11 binds due to the age of both projects. 

Also, i actually remember testing this entire thing with c++11 and the plugin complained IIRC, testing again now though to be sure

Comment (ahmed-elsaharti) [reply CMakeLists.txt:28]: I stand corrected, c++11 seems to work fine. Shall i switch over? @stevenmorphet-at

Comment (stevenmorphet-at) [reply CMakeLists.txt:28]: For now, no. I'd like others to weigh in on what should be done that way.

File: include/livox_laser_simulation/csv_reader.hpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#ifndef SRC_GAZEBO_CSV_READER_HPP
+#define SRC_GAZEBO_CSV_READER_HPP
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+class CsvReader {
```
Comment (stevenmorphet-at) [inline include/livox_laser_simulation/csv_reader.hpp:17]: Maybe add a comment that this is designed to reading a comma separated list of numeric values from file, as opposed to reading a generic CSV file.

Comment (ahmed-elsaharti) [reply include/livox_laser_simulation/csv_reader.hpp:17]: Does [b898d54](https://github.com/Airtonomy/flight-px4-gazebo/pull/3/commits/b898d54dee789e3e89028c1105df9c3deb808e3f) work?

Comment (stevenmorphet-at) [reply include/livox_laser_simulation/csv_reader.hpp:17]: :+1:

File: src/livox_laser_simulation/livox_points_plugin.cpp [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+void convertDataToRotateInfo(const std::vector<std::vector<double>> &datas, std::vector<AviaRotateInfo> &avia_infos) {
+    avia_infos.reserve(datas.size());
+    double deg_2_rad = M_PI / 180.0;
+    for (auto &data : datas) {
+        if (data.size() == 3) {
+            avia_infos.emplace_back();
+            avia_infos.back().time = data[0];
+            avia_infos.back().azimuth = data[1] * deg_2_rad;
+            avia_infos.back().zenith = data[2] * deg_2_rad - M_PI_2;  //转化成标准的右手系角度
```
Comment (stevenmorphet-at) [inline src/livox_laser_simulation/livox_points_plugin.cpp:37]: I take it that this code has been lifted from third party software?

File: src/livox_laser_simulation/livox_points_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+void LivoxPointsPlugin::Load(gazebo::sensors::SensorPtr _parent, sdf::ElementPtr sdf) {
+
+    std::string px4_directory;
+
+    char * val;
+    val = std::getenv("PX4_DIR");
+    if (val != NULL) {
+        px4_directory = val;
+    } else {
+        px4_directory = std::filesystem::current_path().parent_path().parent_path().parent_path();
```
Comment (stevenmorphet-at) [inline src/livox_laser_simulation/livox_points_plugin.cpp:53]: Hmm -- seem to have an implicit folder structure expected here

File: src/livox_laser_simulation/livox_points_plugin.cpp (hunk 3 above)
Comment (ahmed-elsaharti) [reply src/livox_laser_simulation/livox_points_plugin.cpp:37]: Yup, the lib lives in https://github.com/Livox-SDK/livox_laser_simulation

File: src/livox_laser_simulation/livox_points_plugin.cpp (hunk 4 above)
Comment (ahmed-elsaharti) [reply src/livox_laser_simulation/livox_points_plugin.cpp:53]: So the current expectation is we're either:
1. running this through flight-gazebo-sim, where i'll have a PR that exports the px4 autopilot location into an environmental variable that gets parsed and uses in line 51 or
2. We run this through `px4_sitl gazebo_typhoon_h480_livox` where the the env. variable wouldn't be defined resulting in us grabbing the csv from the current run location's parent dir *4.

I thought these would be easier than trying to use cmake to include the csv files into a a lib directory within the build directory or something similar. thoughts?

**Review by stevenmorphet-at — APPROVED** [review]
This code comes from an existing, non-Thread codebase. The amount of Thread code in this PR is pretty small. The non-Thread code is, for me, an undiscovered country. While I do not feel qualified to do a proper review of it, I would assume (famous last words) that the code has been heavily exercised by many developers outside of Thread. Additionally, as this is not code that will be deployed, I am willing to approve it.

Please link an issue to this PR and update the README so that others know that testing has been manually undertaken in simulation.

**Pull request merged**